### PR TITLE
fix(secrev): close the Error/bug tier — tags without a release, and the remains of a 2021 bridge

### PR DESCRIPTION
**Pull request by davidp57**
`SECREV-2` ticket 07, the final Error/bug pass: the last **8 findings, each decided by David** rather than by a sweep, because each needed a call rather than a fix. **Ticket 04 closes with this PR.**

**122 of 140 decided. Every Critical, High, Medium, Security-flaw, Documentation and Error/bug tier is now closed** — what remains is 18 readability/optimization findings that ticket 07's own policy reserves for files being changed anyway.

| | Outcome | |
|---|---|---|
| VMR-104 | **fixed** | tags reached the remote before the release existed |
| VMR-105 | **fixed** | with 104; its "swallowed failures" half does not reproduce |
| VMR-073 | **fixed** | every request errored, and the branch the finding names was unreachable |
| VMR-130 | **fixed** | the remains of the SLMOD bridge, deleted |
| VMR-129 | **wontfix** | legacy one-shot tool, deleted rather than repaired |
| VMR-083 | **wontfix** | three call sites, all debug traces |
| VMR-088 | deferred | one instance of a 794-site family → `REFACTOR-MARKER-PARSER` |
| VMR-089 | deferred | contract hardening the finding itself calls unreachable |

## VMR-104: the ordering was the bug

`publish` pushed the git tags first and created the release second. So an absent `gh` CLI left `published-v<x>` on the remote pointing at a commit with **no release** — and for a full release, `published-latest` had already been force-moved there too. That is the tag the updater resolves and every "latest" link follows.

`gh` is now checked **before** anything reaches the remote, and the floating tag moves only once the release exists (`_move_latest_tag`, split out of `_publish_with_git_tags`). One test pins the ordering itself, not just the outcome.

**VMR-105's other half does not reproduce**, and saying so matters: `logger.error` raises `typer.Abort`, so a failed push or release creation already aborted the publish — which is exactly what keeps the floating tag where it was. The `git tag -d` that runs unchecked is deliberate (it is how re-publishing a version is allowed) and now sits in one place, `_force_tag`, instead of four copies.

## Ticket 04's last item, closed

`download_asset` read `response.content` whole, so a reply with **no `Content-Length` that never ends** had no bound at all — on a tool that installs and then *runs* what it downloads. Now read in 64 KiB chunks against `_MAX_ASSET_BYTES` = 256 MiB, a number taken from measurement rather than picked: the largest real asset is `published.zip` at **61 MiB** (measured on the published release), and 256 MiB is what `safe_zip.MAX_MEMBER_UNCOMPRESSED_BYTES` already uses, so the two bounds in the codebase agree. Five tests, including an endless response and both sides of the boundary.

With it, ticket 04's three shapes all refuse: updater metadata (VMR-011), archive members (VMR-009), network downloads (this).

## VMR-130: the history made the decision

`monitoredCommands` was filled by `veafRemote.monitorWithSlMod(command, script, …)` — the mission-facing half of the **SLMOD** integration. That API was deleted on **2021-08-24** (`067495be`, *"removed slmod monitoring altogether"*, 103 lines), leaving behind a table nothing could fill, a consumer that could only ever warn, and a `mist.utils.dostring` of arbitrary Lua gated by a password that ships in a public repository.

Removed: `executeRemoteCommand`, `markTextAnalysis`, the `_remote` marker command, `monitoredCommands`, `CommandStarter`, the two orphaned `USE_SLMOD*` flags, and the `veafShortcuts` branch that routed markers there. Remote execution keeps its supported route — `registerRemoteModule` / `executeCommandFromRemote`, the only one the server hook uses. **No documentation to update, because `_remote` was documented nowhere**, which is its own kind of evidence. Its tests are replaced by three asserting the path is gone.

## VMR-073 was unreachable where the finding pointed

`handle_client_connection` returns **nothing**, so `success` was always nil: the failure branch concatenated a nil `res` and raised on *every* request, and the `clients[id]` branch the finding names (with `id` undefined and `clients` never filled) could never run. The response is sent before all of it — which is why the smoke harness worked while its server errored each time.

## What was measured rather than assumed

- **VMR-083** — `veaf.serialize` has three call sites in the whole tree, all debug traces (one commented out). Nothing reloads its output as Lua, which is the finding's entire argument.
- **VMR-088** — the logging refactor is real and works: `Logger:trace` checks the level before formatting, and `veaf.lp` defers serialisation through a `__tostring` metatable. Counted: **726** uses of the lazy helper against **794** pre-formatted `trace`/`debug` calls still in `src/scripts/veaf/`. This site defeats both mechanisms at once, and it is one of 794 — a lot, not a finding.

## Verification

- **18 new tests**, each seen failing before its fix, each group with a control
- 3084 Python tests pass, coverage 80.40% against the 79 floor; 36 Lua suites pass
- ruff / ruff format / mypy clean, `docs-check` and `stylua` clean
- The backlog consistency gate caught a stale PRD scope row for ticket 04 and it is aligned
- New i18n key in both locales

## Still not deployed

The two criticals remain live on the VEAF servers until [src/scripts/Hooks/VEAF-Server-hook.lua](src/scripts/Hooks/VEAF-Server-hook.lua) (2.7.1) is copied to `Saved Games/<server>/Scripts/Hooks/`. Checked while answering David's question about v5 compatibility: the three mission-side functions the hook calls have had the same signatures since v5, and each payload is guarded by an existence test, so a v5 mission is a no-op rather than a crash. The real deployment risk is elsewhere — the pilots list moved to the shared `Saved Games/` root, and without it the hook denies every command, as its own error message says. The PRD records this.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by Sourcery

Ensure release publishing and updater downloads fail closed, remove the dead SLMOD marker bridge, and close out the remaining SECREV-2 Error/bug and integrity items.

Bug Fixes:
- Prevent releases from being published as git tags without a corresponding GitHub release by checking for a usable gh CLI before pushing and only moving the floating latest tag after the release exists.
- Cap updater asset downloads by streaming responses and enforcing a maximum asset size so endless or oversized responses are refused instead of exhausting memory.
- Fix the DCS fiddle server loop so client handler errors are reported without crashing the server loop, rather than raising on every request due to an unreachable branch.

Enhancements:
- Refactor git tag management in the release publisher into reusable helpers and add targeted tests that verify tag push order, prerelease behaviour, and failure handling.
- Align integrity and security backlog documentation to reflect completed tickets, closed tiers, and the remaining readability/optimization work.

Tests:
- Add Python tests that exercise the release publisher under gh absence, release-creation failure, tags-only mode, and successful publish to pin the ordering and error propagation.
- Add Python tests that cover updater asset download capping, including declared oversize, endless responses, and boundary conditions around the size limit.
- Update Lua tests for veafRemote to assert that the removed `_remote` marker path and `monitoredCommands` table are no longer present.

Chores:
- Remove the obsolete SLMOD-based `_remote` marker command path and its associated flags, helpers, and tests, keeping only the supported server-hook remote execution route.
- Delete the unused `dictionaryNormalizer.lua` one-shot tool now superseded by the Python pipeline.
- Bump veaf-tools version to 6.13.73 and update SECREV-2 backlog, PRD, and changelog to record the decisions and closure of ticket 04 and the Error/bug tier.

### CODE REVIEW

File: veaf_build/github.py [hunk 1]
```diff
@@ -69,69 +69,81 @@ def publish(self, package_path: Path, package_hash: str, force: bool = False) ->
...
+        # release creation that failed — left `published-v<x>` on the remote with no release behind
+        # it, and `published-latest` force-moved onto that same commit. Anything resolving the
+        # floating tag then pointed at a release GitHub cannot serve. So: refuse before touching the
+        # remote, and move the floating tag only once the release exists.
+        if self.token and not self._gh_cli_available():
+            logger.warning(
+                "GitHub CLI (gh) not found, so no release can be created: nothing was pushed. "
+                "Install it from https://cli.github.com/ or re-run with --skip-git-tags to push tags only."
+            )
+            return
```
[the remark this answers is not in the document]
Comment (davidp57) [reply veaf_build/github.py:86]: Right, and the message was the defect rather than the condition: `--skip-git-tags` *suppresses* the tags, so advising it to "push tags only" was backwards. The real escape hatch is running without a token (the tags-only mode above), and the warning now says that. The guard itself stays as is — with `--skip-git-tags` there is neither a tag nor a release to produce, so returning early is the same outcome by a clearer route.
_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: .backlog/SECREV-2/tickets/07-low-info-sweep.md [hunk 2]
```diff
@@ -867,3 +867,70 @@ Error/bug that each need a decision — VMR-104/105 (the publish path, and a rel
...
+Deleted: `executeRemoteCommand`, `markTextAnalysis`, the `_remote` marker entry point,
+`monitoredCommands`, `CommandStarter`, the two orphaned `USE_SLMOD*` flags, and the `veafShortcuts`
+branch that routed markers there. Nothing to document, because `_remote` was documented nowhere —
+which is itself telling.
+
+### VMR-104: the ordering was the bug, and the tests pin the order
+
+`publish` pushed tags first, created the release second. So an unusable `gh` left `published-v<x>` on
+the remote with no release, and for a full release **`published-latest` force-moved onto that same
+commit** — the tag the updater and every "latest" link resolve. Now `gh` is checked before anything
```
[the remark this answers is not in the document]
Comment (davidp57) [reply .backlog/SECREV-2/tickets/07-low-info-sweep.md:904]: Fixed — "was force-moved".
_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: .backlog/SECREV-2/tickets/07-low-info-sweep.md [hunk 3]
```diff
@@ -867,3 +867,70 @@ Error/bug that each need a decision — VMR-104/105 (the publish path, and a rel
...
+reaches the remote, and the floating tag moves only after the release exists.
+
+Worth recording: the "swallowed local failures" of VMR-105 does not reproduce. `logger.error` raises
```
[the remark this answers is not in the document]
Comment (davidp57) [reply .backlog/SECREV-2/tickets/07-low-info-sweep.md]: Fixed — plural subject, "do not reproduce".
_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: .backlog/SECREV-2/tickets/04-fail-closed-integrity.md [hunk 4]
```diff
@@ -32,8 +32,18 @@ decompression, and verify what was fetched before using it.
...
+- [x] **Cap and verify the bridge and updater downloads.** Both halves are in, and they landed in
+      different lots, which is why this line stayed open longer than it should have. The *verify*
+      half came with `SECREV-2` ticket 07: the bridge fetch is capped at 2 MiB (VMR-034, measured
+      against its real 13 237 bytes) and every hop of a release-asset download must be https on a
+      GitHub host (VMR-037) — a fix whose first version had the very hole it was closing, since
+      `requests` follows redirects anywhere by default. The *cap* half is this pass:
+      `download_asset` streamed nothing and read `response.content` whole, so a reply with no
+      `Content-Length` that never ends had no bound at all — on an updater that installs and then
+      **runs** what it downloads. Now read in 64 KiB chunks against `_MAX_ASSET_BYTES` = 256 MiB,
+      a bound chosen from measurement (largest real asset: `published.zip` at 61 MiB) and matching
```
[the remark this answers is not in the document]
Comment (davidp57) [reply .backlog/SECREV-2/tickets/04-fail-closed-integrity.md:44]: Fixed — "It now reads in 64 KiB chunks".
_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
